packages/core/src/lib/base_protocol.ts

```diff
@@ -50,15 +59,4 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
-  protected async newStream(peer: Peer): Promise<Stream> {
```

Comment:
I'd leave `newStream` method on `BaseProtocol` as a facade and hide `StreamManager` as part of internal implementation.

Reply:
well, stream manager is more of an optimization than a service so I don't see a need to pass this complexity to consumers

Reply:
even with this perspective given the only change is:
```ts
this.newStream -> this.streamManager.getStream
```
we might as well just keep `this.newStream` as facade

Reply:
ok, `getStream` fairer than `newStream`